File: LeetCode-CPP/206.反转链表.java
/*
 * @lc app=leetcode.cn id=206 lang=java
 *
 * [206] 反转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode last = null;
        //改每条边，所以要访问链表
        while(head != null){
            ListNode nextHead = head.next;
            //改一条边
            head.next = last;
            //last，head向后移动一位
            last = head;
            head = nextHead;
        }
        return last;
    }
}
// @lc code=end

File: LeetCode-C++/Linked List/206.反转链表.cpp
/*
 * @lc app=leetcode.cn id=206 lang=cpp
 *
 * [206] 反转链表
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head == nullptr) {
            return head;
        }
        ListNode *node = nullptr;
        ListNode *cur = head;
        while(cur != nullptr) {
            ListNode *tail = cur->next;
            cur->next = node;
            node = cur;
            cur = tail;
        }
        return node;
    }
};
// @lc code=end

